src/backend/joanie/payment/backends/base.py

```diff
@@ -168,13 +172,65 @@ def _send_mail_payment_installment_success(
                 to_user_email=order.owner.email,
             )
 
-    @staticmethod
-    def _do_on_payment_failure(order, installment_id):
+    @classmethod
+    def _send_mail_refused_debit(cls, order, installment_id):
+        """
+        Prepare mail context when debit has been refused for an installment in the
+        the current language of the user.
+        """
+        with override(order.owner.language):
+            title = order.product.safe_translation_getter(
+                "title", language_code=order.owner.language
+            )
+            amount = Money(
+                next(
+                    installment["amount"]
+                    for installment in order.payment_schedule
+                    if installment["id"] == installment_id
+                ),
+                currency=settings.DEFAULT_CURRENCY,
+            )
+            position = order.get_position_of_last_installment(
+                state=PAYMENT_STATE_REFUSED
+            )
+            nth_installment = position + 1  # position index starts at 0
```

Comment:
I wonder if we should do that here or directly in the template ?

Reply:
Yeah, I know this is annoying to add + 1 here. I agree with you.

We made the index to start at 0 with enumerate, (in this method : `def get_index_of_last_installment(self, state):)`  and in the forloop counter in the template MJML, we've set it to start at index 0. It worked perfectly for `paid` installments, but for "refused" installment, we needed to add +1.

I can either change the enumerate to start at 1, instead of 0. With this change, I can also change in my `installment_table.mjml` the forloop0 to a standard forloop that starts at 1. That's a suggestion that I need to verify, in making sure that it will work in every cases (`paid`, `pending` and `refused` states)